logstash-core/lib/logstash/config/pipeline_config.rb

```diff
@@ -44,5 +44,19 @@ def display_debug_information
       logger.debug("Merged config")
       logger.debug("\n\n#{config_string}")
     end
+
+    def lookup_source_and_line(merged_config_line)
+      remaining_lines = merged_config_line
+      matching_part = nil
+      @config_parts.each do |source_with_meta|
+        if remaining_lines <= source_with_meta.lines_count
```

Comment:
nitpick: cache the result in a local variable to avoid having to calculate the line count twice per config part

```suggestion
        part_lines_count = source_with_meta.getLinesCount()
        if remaining_lines <= parts_lines_count
```

Note that the above uses the java method name per [this comment](https://github.com/elastic/logstash/pull/11155/files#diff-71eb42b2b4da64252eedcf2a21d7e0e8R43) and has a related change in [this comment](https://github.com/elastic/logstash/pull/11155/files#diff-a9a79602308f5b7e16ce16d13287b75eR56)